File: modules/core-api/src/main/java/com/enonic/xp/core/security/RoleKeys.java
package com.enonic.xp.core.security;

public final class RoleKeys
{
    private RoleKeys()
    {
    }

    public static final PrincipalKey ADMIN = PrincipalKey.ofRole( "system.admin" );

    public static final PrincipalKey EVERYONE = PrincipalKey.ofRole( "system.everyone" );

    public static final PrincipalKey AUTHENTICATED = PrincipalKey.ofRole( "system.authenticated" );

    public static final String ADMIN_LOGIN_ID = "system.admin.login";

    public static final PrincipalKey ADMIN_LOGIN = PrincipalKey.ofRole( ADMIN_LOGIN_ID );

    public static final PrincipalKey USER_MANAGER_APP = PrincipalKey.ofRole( "system.user.app" );

    public static final PrincipalKey USER_MANAGER_ADMIN = PrincipalKey.ofRole( "system.user.admin" );

    public static final PrincipalKey CONTENT_MANAGER_APP = PrincipalKey.ofRole( "cms.cm.app" );

    public static final PrincipalKey CONTENT_MANAGER_ADMIN = PrincipalKey.ofRole( "cms.admin" );

}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/security/RoleKeys.ts
module api.security {

    export class RoleKeys {

        private static ROLE_ADMIN: string = 'system.admin';

        private static ROLE_CMS_ADMIN: string = 'cms.admin';

        private static ROLE_USER_ADMIN: string = 'system.user.admin';

        private static ROLE_EVERYONE: string = 'system.everyone';

        private static ROLE_AUTHENTICATED: string = 'system.authenticated';

        private static ROLE_CMS_EXPERT: string = 'cms.expert';

        /* */

        public static EVERYONE: PrincipalKey = PrincipalKey.ofRole(RoleKeys.ROLE_EVERYONE);

        public static AUTHENTICATED: PrincipalKey = PrincipalKey.ofRole(RoleKeys.ROLE_AUTHENTICATED);

        public static ADMIN: PrincipalKey = PrincipalKey.ofRole(RoleKeys.ROLE_ADMIN);

        public static CMS_ADMIN: PrincipalKey = PrincipalKey.ofRole(RoleKeys.ROLE_CMS_ADMIN);

        public static USER_ADMIN: PrincipalKey = PrincipalKey.ofRole(RoleKeys.ROLE_USER_ADMIN);

        public static CMS_EXPERT: PrincipalKey = PrincipalKey.ofRole(RoleKeys.ROLE_CMS_EXPERT);

        /* */

        private static contentAdminRoles: string[] = [RoleKeys.ROLE_ADMIN, RoleKeys.ROLE_CMS_ADMIN];

        private static userAdminRoles: string[] = [RoleKeys.ROLE_ADMIN, RoleKeys.ROLE_USER_ADMIN];

        private static contentExpertRoles: string[] = [RoleKeys.ROLE_ADMIN, RoleKeys.ROLE_CMS_ADMIN, RoleKeys.ROLE_CMS_EXPERT];

        /* */

        public static isContentAdmin(principalKey: PrincipalKey): boolean {
            return RoleKeys.contentAdminRoles.some(roleId => principalKey.getId() === roleId);
        }

        public static isUserAdmin(principalKey: PrincipalKey): boolean {
            return RoleKeys.userAdminRoles.some(roleId => principalKey.getId() === roleId);
        }

        public static isContentExpert(principalKey: PrincipalKey): boolean {
            return RoleKeys.contentExpertRoles.some(roleId => principalKey.getId() === roleId);
        }

        public static isAdmin(principalKey: PrincipalKey): boolean {
            return RoleKeys.ROLE_ADMIN === principalKey.getId();
        }

        public static isEveryone(principalKey: PrincipalKey): boolean {
            return RoleKeys.ROLE_EVERYONE === principalKey.getId();
        }
    }
}
